### PR TITLE
feat(containers): Support async multisection datalist

### PR DESCRIPTION
**Pull request by tlnd-mhuchet**
**What is the problem this PR is trying to solve?**
Multisection datalist are not supported in `resolveNameFortitleMap` because the `titleMap` has a different shape.

In "classic" datalist, the `titleMap` is an array of objects with this shape:
```javascript
{ name: "Readable label", value: "someUglyId" }
```

In a multisection datalist, the `titleMap` is an array of objects with this shape:
```javascript
{
  title: "Section title",
  suggestions: [
    { name: "Readable label", value: "someUglyId" },
    { name: "Another label", value: "someUglyId2" },
    ...
  ]
}
```

**What is the chosen solution to this problem?**
Add the support of multisection datalist by checking the presence of `suggestions`. If we find it, we assume it is a multisection and handle it. If not we keep the current behavior for classic datalists.

**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Review by jmfrancois — COMMENTED** [review]
please add a unit test

**Comment by tlnd-mhuchet** [comment]
> please add a unit test

Not one, but three! 😄 
Tried to cover the sames cases as we do for no multi section lists.

**Review by romainseb — APPROVED** [review]

File: packages/containers/src/ComponentForm/ComponentForm.component.js [hunk 1]
```diff
@@ -46,15 +46,25 @@ export function resolveNameForTitleMap({ schema, properties, value }) {
...
-	const names = uniformValue
-		.map(nextValue => schema.titleMap.find(titleMap => titleMap.value === nextValue))
-		.map(entry => entry && entry.name);
+	const names = schema.titleMap
+		.map(item => {
+			if (item.suggestions) {
+				const suggestionItem = item.suggestions.find(subItem =>
+					uniformValue.includes(subItem.value),
+				);
+				return suggestionItem ? suggestionItem.name : undefined;
```
Comment (gjou-tlnd) [inline packages/containers/src/ComponentForm/ComponentForm.component.js:60]:
```suggestion
				return suggestionItem?.name;
```

File: packages/containers/src/ComponentForm/ComponentForm.test.js [hunk 2]
```diff
@@ -117,7 +117,7 @@ describe('ComponentForm', () => {
 			// then
 			expect(properties.my.awesome).toEqual({
 				value: ['correct value', 'neither'],
-				$value_name: ['Yes this is the name', 'Neither this one'],
+				$value_name: ['Neither this one', 'Yes this is the name'],
```
Comment (jmfrancois) [inline packages/containers/src/ComponentForm/ComponentForm.test.js:120]: why value name is reversed compared to the value ?

**Review by jmfrancois — CHANGES_REQUESTED** [review]
the order of value name seems wrong

File: packages/containers/src/ComponentForm/ComponentForm.test.js [hunk 3]
```diff
@@ -41,7 +41,7 @@ describe('ComponentForm', () => {
 		});
 	});
 
-	describe('#resolveNameForTitleMap', () => {
+	describe.only('#resolveNameForTitleMap', () => {
```
Comment (tlnd-mhuchet) [inline packages/containers/src/ComponentForm/ComponentForm.test.js:44]:
```suggestion
	describe('#resolveNameForTitleMap', () => {
```

File: packages/containers/src/ComponentForm/ComponentForm.test.js (hunk 2 above)
Comment (tlnd-mhuchet) [reply packages/containers/src/ComponentForm/ComponentForm.test.js]: Yes you are right. Good catch.

**Comment by tlnd-mhuchet** [comment]
> the order of value name seems wrong

Should be better now!

File: packages/containers/src/ComponentForm/ComponentForm.component.js [hunk 4]
```diff
@@ -46,14 +46,31 @@ export function resolveNameForTitleMap({ schema, properties, value }) {
...
-
 	// Here we add a field side by side with the value
 	// to keep the title associated to the value
 	const valueIsArray = Array.isArray(value);
 	const uniformValue = valueIsArray ? value : [value];
 
 	const names = uniformValue
-		.map(nextValue => schema.titleMap.find(titleMap => titleMap.value === nextValue))
+		.map(nextValue => {
+			const titleMap = schema.titleMap;
```
Comment (sgendre) [inline packages/containers/src/ComponentForm/ComponentForm.component.js:56]: please use spread operator

Comment (tlnd-mhuchet) [reply packages/containers/src/ComponentForm/ComponentForm.component.js:56]: Don't be shy, use suggestions 😄 

```suggestion
			const { titleMap } = schema;
```

File: packages/containers/src/ComponentForm/ComponentForm.component.js [hunk 5]
```diff
@@ -46,14 +46,31 @@ export function resolveNameForTitleMap({ schema, properties, value }) {
...
 	// Here we add a field side by side with the value
 	// to keep the title associated to the value
 	const valueIsArray = Array.isArray(value);
 	const uniformValue = valueIsArray ? value : [value];
 
 	const names = uniformValue
-		.map(nextValue => schema.titleMap.find(titleMap => titleMap.value === nextValue))
+		.map(nextValue => {
+			const { titleMap } = schema;
+			const isMultiSection = !!titleMap[0]?.suggestions;
```
Comment (jmfrancois) [inline packages/containers/src/ComponentForm/ComponentForm.component.js:57]: please move those two var outside of the map

**Review by jmfrancois — APPROVED** [review]

**Pull request merged**